classes/UpgradeTools/FilesystemAdapter.php

```diff
@@ -202,56 +202,37 @@ public function listSampleFiles($dir, $fileext = '.jpg')
      * @param type $file : current file or directory name eg:'.svn' , 'settings.inc.php'
      * @param type $fullpath : current file or directory fullpath eg:'/home/web/www/prestashop/app/config/parameters.php'
      * @param type $way : 'backup' , 'upgrade'
+     * @param string $temporaryWorkspace : If needed, another folder than the shop root can be used (used for releases)
      */
-    public function isFileSkipped($file, $fullpath, $way = 'backup')
+    public function isFileSkipped($file, $fullpath, $way = 'backup', $temporaryWorkspace = null)
     {
         $fullpath = str_replace('\\', '/', $fullpath); // wamp compliant
-        $rootpath = str_replace('\\', '/', $this->prodRootDir);
-        switch ($way) {
-            case 'backup':
-                if (in_array($file, $this->fileFilter->getExcludeFiles())) {
-                    return true;
-                }
-
-                foreach ($this->fileFilter->getFilesToIgnoreOnBackup() as $path) {
-                    $path = str_replace(DIRECTORY_SEPARATOR . 'admin', DIRECTORY_SEPARATOR . $this->adminSubDir, $path);
-                    if ($fullpath === $rootpath . $path) {
-                        return true;
-                    }
-                }
-                break;
-                // restore or upgrade way : ignore the same files
-                // note the restore process use skipFiles only if xml md5 files
-                // are unavailable
-            case 'restore':
-                if (in_array($file, $this->fileFilter->getExcludeFiles())) {
-                    return true;
-                }
+        if (null !== $temporaryWorkspace) {
+            $rootpath = str_replace('\\', '/', $temporaryWorkspace);
+        } else {
+            $rootpath = str_replace('\\', '/', $this->prodRootDir);
```

Comment:
```php
$rootpath = str_replace('\\', '/', null !== $temporaryWorkspace ? $temporaryWorkspace : $this->prodRootDir);
```

Reply:
I chose to not write as you suggested. This makes the line quite difficult to read.

Reply:
And like this ? :D

```php
$rootpath = str_replace(
    '\\', 
    '/', 
    null !== $temporaryWorkspace ? $temporaryWorkspace : $this->prodRootDir
);
```

Reply:
Meh, ok!